src/core/agent/services/connectionService.ts

```diff
@@ -123,6 +124,24 @@ class ConnectionService extends AgentService {
     return connectionRecord.state === DidExchangeState.Completed;
   }
 
+  async resolveOObi(url: string, name: string): Promise<void> {
+    const resolvedOobi = await this.agent.modules.signify.resolveOobi(url);
+    let resolvedOobis: Record<string, any> = {};
+    try {
+      const storedResolvedOobis = await PreferencesStorage.get(
+        PreferencesKeys.APP_TUNNEL_CONNECT
+      );
+      resolvedOobis = storedResolvedOobis || {};
+    } catch (e) {
+      // TODO: handle error
+    }
+
+    resolvedOobis[name] = resolvedOobi;
+    await PreferencesStorage.set(
+      PreferencesKeys.APP_TUNNEL_CONNECT,
+      resolvedOobis
+    );
```

Comment:
I wonder if it's better to store something more like:

`resolvedOobis[oobiAID] = { name, url }` - I'm not sure if you need to store the entire resolved OOBI, all that is stored in KERIA.

And the reason I would put `oobiAID` as the key is that in your next story, when you receive an exn message from `oobiAID` - you can quickly check if `resolvedOobis[oobiAID]` is undefined.

Reply:
So far, both in the extension and in the wallet, I am keeping the whole answer to solve the OOBI to reduce the calls to Keria in case I need extra information.
I will change the key to `response.i`

Reply:
The OOBI is only a discovery mechanism to bootstrap a secure communication for a contact to AID X. So that when you send any message to AID X, it knows where to go. So I don't think there's anything in the OOBI response you need once it's fully resolved.

Reply:
Okey, so i will keep it as `resolvedOobis[oobiAID] = { name, url }`